sway/tree/container.c

```diff
@@ -349,7 +349,7 @@ static bool surface_is_popup(struct wlr_surface *surface) {
 	if (wlr_surface_is_xdg_surface_v6(surface)) {
 		struct wlr_xdg_surface_v6 *xdg_surface_v6 =
 			wlr_xdg_surface_v6_from_wlr_surface(surface);
-		while (xdg_surface_v6) {
+		while (xdg_surface_v6 && xdg_surface_v6->role == WLR_XDG_SURFACE_V6_ROLE_NONE) {
```

Comment:
This is wrong. It should be `!=`, or `== WLR_XDG_SURFACE_ROLE_TOPLEVEL` as mentioned in another comment.